src/screens/stationSelect.tsx

```diff
@@ -58,6 +56,8 @@ const StationSelectScreen: React.FC<StationSelectProps> = ({ navigation }) => {
   const [arriveStationCrsCode, setArriveStationCrsCode] = useState<
     string | null
   >(null);
+  const [showErrorMessage, setShowErrorMessage] = useState<boolean>(false);
```

Comment:
I'm not sure you need this `useState` hook - is there ever a time you want to set an error message and not show it? If not, you can just control whether you show it or not based on if the other hook is null (and then to hide it just set the message as null).

That means you only have to worry about one bit of state and never risk accidentally putting your two states out of sync with each other.